dlrover/python/master/node/dist_job_manager.py

```
@@ -1019,6 +1053,47 @@ def _relaunch_node(self, node: Node):
         self._job_context.update_job_node(node)
         self._scaler.scale(plan)
 
+    def _relaunch_node_group(self, node_group: int):
+        """
+        Relaunch all nodes in the node_group group, no matter it is
+        running or pending.
+        The group index should be different with all existing group,
+        so we use max_group_idx to keep it in record
+
+        """
+        self._job_context.max_group_idx += 1
+        logger.info(
+            f"Relaunch node group {node_group} with "
+            f"group_idx {self._job_context.max_group_idx}"
+        )
+
+        launch_nodes: List[Node] = []
+        for node in self._job_context.job_node_group(node_group).values():
+            if node.type != NodeType.WORKER:
+                logger.error(f"Fatal error: {node.name} is not worker node")
```

why fatal error here? if u ignore this error, u shouldn't call it fatal

---

just warning logging and break the loop